src/keyRotation/keyRotator.test.ts

```diff
@@ -128,17 +128,20 @@ test('2 inactive keys', (done) => {
 
     keyRotator.rotateKeys(user)
         .then(() => {
-            fail();
-            done();
-        })
-        .catch((err) => {
-            // Expect there to be 2 keys
-            expect(keys.length).toBe(0);
+            // Expect there to be 1 key
+            expect(keys.length).toBe(1);
 
-            // Expect both existing keys to have been removed
+            // Expect existing keys to have been removed
             expect(keys.indexOf(firstExistingKey)).toBe(-1);
             expect(keys.indexOf(secondExistingKey)).toBe(-1);
 
+            // Expect new key to be present and active
+            expect(newKey.Status).toBe(ACTIVE);
+            expect(keys.indexOf(newKey)).toBeGreaterThanOrEqual(0);
+            done(); done();
+        })
+        .catch((err) => {
+            fail();
             done();
```

Comment:
Do you need a `done` when there's a `fail`?

Reply:
Also, I believe you can return a promise instead of using `done()` and `fail()`

Reply:
I don't think the `done`s are required after the `fail`s, either way the test would fail if it hit that path.

Not sure about returning a promise instead of `done()`, I can't seem to get the tests working without the `done`s as they just timeout.